system-addon/lib/TelemetryFeed.jsm

```diff
@@ -242,6 +309,36 @@ this.TelemetryFeed = class TelemetryFeed {
     this.telemetrySender.sendPing(await eventPromise);
   }
 
+  async handleImpressionStats(action) {
```

Comment:
I don't think we need async here. There's no promise returned or long running call that we're awaiting.